===>  Building for ppsspp-qt5-1.4.2
ninja: error: dependency cycle: cmake_object_order_depends_target_Core -> Core_autogen -> CMakeFiles/Core_autogen -> /wrkdirs/usr/ports/emulators/ppsspp-qt5/work/ppsspp-1.4.2/git-version.cpp -> cmake_object_order_depends_target_Core

https://github.com/hrydgard/ppsspp/issues/9972

--- CMakeLists.txt.orig	2017-05-27 09:58:05 UTC
+++ CMakeLists.txt
@@ -1618,12 +1618,11 @@ endif()
 
 setup_target_project(${CoreLibName} Core)
 
-# Generate git-version.cpp at build time.
-add_custom_target(GitVersion ALL
-	DEPENDS something_that_never_exists)
-add_custom_command(OUTPUT something_that_never_exists
+add_custom_command(OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/git-version.cpp
 	COMMAND ${CMAKE_COMMAND} -DSOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}
 		-P ${CMAKE_CURRENT_SOURCE_DIR}/git-version.cmake)
+# Generate git-version.cpp at build time.
+add_custom_target(GitVersion DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/git-version.cpp)
 
 set_source_files_properties(${CMAKE_CURRENT_SOURCE_DIR}/git-version.cpp
 	PROPERTIES GENERATED TRUE)
